OPAL/tac/src/main/scala/org/opalj/tac/cg/android/AndroidManifestKey.scala

```diff
@@ -0,0 +1,104 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package tac
+package cg
+package android
+
+import org.opalj.br.ClassFile
+import org.opalj.br.ObjectType
+import org.opalj.br.analyses.ProjectInformationKey
+import org.opalj.br.analyses.ProjectInformationKeys
+import org.opalj.br.analyses.SomeProject
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.xml.Elem
+
+/**
+ * A ProjectInformationKey that is used to parse an AndroidManifest.xml. It returns a map of Android components and
+ * a ListBuffer of all IntentFilters defined in the manifest.
+ * The AndroidManifest.xml can be set as initialization data.
+ *
+ * @author Tom Nikisch
+ */
+object AndroidManifestKey extends ProjectInformationKey[Option[(Map[String, ListBuffer[ClassFile]], List[IntentFilter])], Elem] {
+
+    override def requirements(project: SomeProject): ProjectInformationKeys = Nil
+
+    override def compute(project: SomeProject): Option[(Map[String, ListBuffer[ClassFile]], List[IntentFilter])] = {
```

Comment:
Not a nice type at all. A map with fixed string keys instead of a tuple/case class, mutable list buffer, etc.